controllers/controllers/workloads/cfspace_controller.go

```diff
@@ -324,14 +325,22 @@ func (r *CFSpaceReconciler) finalize(ctx context.Context, log logr.Logger, space
 		return ctrl.Result{}, nil
 	}
 
-	err := r.finalizeCFApp(ctx, log, space.GetName())
-	if err != nil {
-		log.Error(err, "failed to finalize CFApps while deleting CFSpace")
+	duration := time.Since(space.GetDeletionTimestamp().Time)
+	log.Info(fmt.Sprintf("finalizing CFSpace for %fs", duration.Seconds()))
+	if duration < 60.0*time.Second {
+		err := r.finalizeCFApps(ctx, log, space.GetName())
+		if err != nil {
+			log.Error(err, "failed to finalize CFApps while deleting CFSpace")
+			return ctrl.Result{RequeueAfter: 500 * time.Millisecond}, nil
```

Comment:
What's the behavior when deleting a cfapp gets stuck? I think eventually we want to start returning errors here, and *maybe* also go ahead and delete the namespace, since that will clean up a lot of other resources.

Reply:
So if a CFApp gets stuck, eventually the time since calculation using the space deletion timestamp will be greater than 60s and it will fall into the else case and continue on to delete the namespace. At that point it will stop trying to finalize the CFApps.